features/attached_enable.feature

```diff
@@ -672,6 +672,45 @@ Feature: Enable command behaviour when attached to an Ubuntu Pro subscription
            | xenial  |
            | bionic  |
 
+    @series.xenial
+    @series.bionic
+    @uses.config.machine_type.lxd.vm
+    Scenario Outline: Attach works when snapd cannot be installed
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I run `apt-get remove -y snapd` with sudo
+        And I create the file `/etc/apt/preferences.d/no-snapd` with the following
+        """
+        Package: snapd
+        Pin: release o=*
+        Pin-Priority: -10
+        """
+        And I run `apt-get update` with sudo
+        When I attempt to attach `contract_token` with sudo
+        Then I will see the following on stderr:
+        """
+        Enabling default service esm-apps
+        Enabling default service esm-infra
+        Enabling default service livepatch
+        Failed to enable default services, check: sudo pro status
+        """
+        When I run `pro status` with sudo
+        Then stdout matches regexp:
+        """
+        livepatch +yes +disabled
+        """
+        Then I verify that running `pro enable livepatch` `with sudo` exits `1`
+        Then I will see the following on stdout:
+        """
+        One moment, checking your subscription first
+        Installing snapd
+        Updating package lists
+        Failed to install snapd on the system
+        """
+        Examples: ubuntu release
+           | release |
+           | xenial  |
+           | bionic  |
```

Comment:
should we also run this on focal or jammy?

Reply:
I ddin't add focal/jammy because livepatch is not supported for the kvm kernel there, and without your new changes it wouldn't work. Other livepatch tests in this file also rely only on Xenial/Bionic, and it covers well the bugfix imho. Do you think all LTS should be included?

Reply:
Good points. I think it is good as it is then!